### PR TITLE
fontconfig: modernize + msvc support + bump deps

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **fontconfig/all**

- add msvc
- modernize
- bump deps

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SSE4** [comment]
need to remove PDB files

File: recipes/fontconfig/all/conanfile.py [hunk 1]
```diff
@@ -1,109 +1,172 @@
...
+                meson = self._configure_meson()
+                meson.install()
+                if os.path.isfile(os.path.join(self.package_folder, "lib", "libfontconfig.a")):
+                    tools.rename(os.path.join(self.package_folder, "lib", "libfontconfig.a"),
+                                 os.path.join(self.package_folder, "lib", "fontconfig.lib"))
+        else:
+            with tools.run_environment(self):
+                autotools = self._configure_autotools()
+                autotools.install()
+        tools.remove_files_by_mask(os.path.join(self.package_folder, "bin", "etc", "fonts", "conf.d"), "*.conf")
```
Comment (madebr) [inline recipes/fontconfig/all/conanfile.py:156]:
```suggestion
        tools.remove_files_by_mask(os.path.join(self.package_folder, "bin"), "*.pdb")
        tools.remove_files_by_mask(os.path.join(self.package_folder, "bin", "etc", "fonts", "conf.d"), "*.conf")
```

Comment (ericLemanissier) [reply recipes/fontconfig/all/conanfile.py:157]: thanks @SSE4 😉

Comment (madebr) [reply recipes/fontconfig/all/conanfile.py:157]: Oh, then thanks to both of you!! 😄 😉

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by sh0 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**